Source/Core/Core/HW/DVD/DVDThread.cpp

```diff
@@ -342,20 +342,34 @@ static void FinishRead(u64 id, s64 cycles_late)
             (CoreTiming::GetTicks() - request.time_started_ticks) /
                 (SystemTimers::GetTicksPerSecond() / 1000000));
 
+  DVDInterface::DIInterruptType interrupt;
   if (buffer.size() != request.length)
   {
-    PanicAlertT("The disc could not be read (at 0x%" PRIx64 " - 0x%" PRIx64 ").",
-                request.dvd_offset, request.dvd_offset + request.length);
+    if (request.dvd_offset != 0x118280000 && request.dvd_offset != 0x1FB500000)
+    {
+      PanicAlertT("The disc could not be read (at 0x%" PRIx64 " - 0x%" PRIx64 ").",
+                  request.dvd_offset, request.dvd_offset + request.length);
+    }
+    else
+    {
+      // Part of the error 001 check.
+      INFO_LOG(DVDINTERFACE, "Ignoring out of bounds test read (at 0x%" PRIx64 " - 0x%" PRIx64 ")",
+               request.dvd_offset, request.dvd_offset + request.length);
+    }
+
+    DVDInterface::SetHighError(DVDInterface::ERROR_BLOCK_OOB);
```

Comment:
That we have reached this place in the code doesn't necessarily mean that we attempted to read out of bounds. We can also reach this under conditions such as having the game stored on a flash drive and yanking the flash drive out in the middle of gameplay. If you want to check whether we read out of bounds, you probably want to make use of this:

https://github.com/dolphin-emu/dolphin/blob/19ed64122c11f7cd36ff92b61e7af7bb772c01a8/Source/Core/DiscIO/Volume.h#L131

But keep in mind that it usually returns a too large value for WBFS and CISO as a result of those formats not containing size information. (The current approach has the same problem, though. Reads won't fail due to out of bounds when reading within the bounds of what `GetSize()` returns.)

Reply:
I guess I probably should reply to this: I don't plan on addressing this at the moment.  The main benefit of the change is that it returns an error to the game when an actual error occurs; making it match the actual error would be ideal but would require more complex changes to actually identify what went wrong.

I'll probably look into it for a separate pull request (probably along with completely async reads where it can wait longer than normal as an option).